pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,257 @@
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_("Private key")} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} className='pf-v5-u-font-family-monospace' value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
+                    <Button variant='secondary'
+                        onClick={async () => {
+                            try {
+                                const key = await cockpit.spawn(["wg", "genkey"]);
+                                setPrivateKey(key.trim());
+                            } catch (e) {
+                                setDialogError(e.message);
+                            }
+                        }}
+                    >
+                        {_("Generate")}
+                    </Button>
+                </InputGroup>
+            </FormGroup>
+            <FormGroup label={_("Public key")}>
+                <ClipboardCopy isReadOnly className='pf-v5-u-font-family-monospace'>{publicKey}</ClipboardCopy>
+            </FormGroup>
+            <FormGroup label={_("Listen port")} fieldId={idPrefix + '-listen-port-input'}>
+                <TextInput id={idPrefix + '-listen-port-input'} placeholder={_("Random port selected when unspecified")} type='number' value={listenPort} onChange={(_, val) => { setListenPort(val) }} />
+            </FormGroup>
+            <FormGroup label={_("Addresses")} fieldId={idPrefix + '-addresses-input'}>
+                <TextInput id={idPrefix + '-addresses-input'} value={addresses} onChange={(_, val) => { setAddresses(val) }} placeholder={_("E.g. 10.0.0.1/24")} />
+            </FormGroup>
+            <FormFieldGroup
+                header={
+                    <FormFieldGroupHeader
+                        titleText={{ text: _("Peers") }}
+                        actions={
+                            <Button
+                                variant='secondary'
+                                onClick={() => setPeers(peers => [...peers, { publicKey: '', endpoint: '', allowedIps: '' }])}
+                            >
+                                {_("Add peer")}
+                            </Button>
+                        }
+                    />
+                }
+                className='dynamic-form-group'
+            >
+                {(peers.length !== 0)
+                    ? peers.map((peer, i) => (
+                        <Grid key={i} hasGutter>
+                            <FormGroup className='pf-m-6-col-on-md' label={_("Public key")} fieldId={idPrefix + '-publickey-peer-' + i}>
+                                <TextInput
+                                    value={peer.publicKey}
+                                    onChange={(_, val) => {
+                                        setPeers(peers => peers.map((peer, index) => i === index ? { ...peer, publicKey: val } : peer));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230719-160604-4ed0b4ac-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)